### PR TITLE
Core/Loot: Implement CMSG_LOOT_MONEY, CMSG_LOOT_RELEASE, SMSG_COIN_REMOVED, SMSG_LOOT_MONEY_NOTIFY

### PR DESCRIPTION
**Pull request by cemak**

### CODE REVIEW

File: src/server/game/Server/Packets/LootPackets.cpp [hunk 1]
```diff
@@ -85,3 +85,23 @@ WorldPacket const* WorldPackets::Loot::LootRemoved::Write()
...
+
+void WorldPackets::Loot::LootRelease::Read()
+{
+    _worldPacket >> Unit;
+}
+
+WorldPacket const* WorldPackets::Loot::LootMoneyNotify::Write()
+{
+    _worldPacket << Money;
+    _worldPacket.WriteBit(SoleLooter);
```
Comment (DDuarte) [inline src/server/game/Server/Packets/LootPackets.cpp]: This might need a FlushBits after WriteBit

Comment (cemak) [inline src/server/game/Server/Packets/LootPackets.cpp]: make

Comment (Warpten) [inline src/server/game/Server/Packets/LootPackets.cpp]: s/might/does

**Comment by cemak** [comment]
[URL=http://www.radikal.ru][IMG]http://s017.radikal.ru/i425/1501/8f/f97b0a12e87b.jpg[/IMG][/URL]

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -8944,10 +8944,11 @@ void Player::SendLootError(ObjectGuid guid, LootError error)
...
 }
 
-void Player::SendNotifyLootMoneyRemoved()
+void Player::SendNotifyLootMoneyRemoved(ObjectGuid lootObj)
 {
-    WorldPacket data(SMSG_LOOT_CLEAR_MONEY, 0);
-    GetSession()->SendPacket(&data);
+    WorldPackets::Loot::CoinRemoved packet;
+    packet.LootObj = lootObj;
+    GetSession()->SendPacket(packet.Write());
```
Comment (Intel) [inline src/server/game/Entities/Player/Player.cpp]: It's better to use `Player::SendDirectMessage(WorldPacket const* data)`

File: src/server/game/Handlers/LootHandler.cpp [hunk 3]
```diff
@@ -196,10 +196,10 @@ void WorldSession::HandleLootMoneyOpcode(WorldPacket& /*recvData*/)
...
                         guild->HandleMemberDepositMoney(this, guildGold, true);
 
-                WorldPacket data(SMSG_LOOT_MONEY_NOTIFY, 4 + 1);
-                data << uint32(goldPerPlayer);
-                data << uint8(playersNear.size() <= 1); // Controls the text displayed in chat. 0 is "Your share is..." and 1 is "You loot..."
-                (*i)->GetSession()->SendPacket(&data);
+                WorldPackets::Loot::LootMoneyNotify packet;
+                packet.Money = goldPerPlayer;
+                packet.SoleLooter = playersNear.size() <= 1 ? true : false;
+                (*i)->GetSession()->SendPacket(packet.Write());
```
Comment (Intel) [inline src/server/game/Handlers/LootHandler.cpp]: Same as above `Player::SendDirectMessage(WorldPacket const* data)`

**Comment by Intel** [comment]
Other than those everything seems OK.
Squash everything and we will merge

**Comment by cemak** [comment]
Intel, all

**Comment by Intel** [comment]
Merged in https://github.com/TrinityCore/TrinityCore/commit/44b2e520218f3d0d1969efb7fa22c869e5c7833f

Something was wrong with merge message after I squashed commits

**Pull request closed without merging**